### PR TITLE
Add Travis Build Stages and CircleCI recipes

### PR DESCRIPTION
**Pull request by pvdlg**
Fix #573

### CODE REVIEW

File: docs/recipes/travis-build-stages.md [hunk 1]
```diff
@@ -1 +1,114 @@
...
+  - 4
+
+os:
+  - linux
+  - osx
+
+# Run the default test stage first, then the release one only when all the test jobs are successful
+stages:
+  - test
+  - release
```
Comment (gr2m) [inline docs/recipes/travis-build-stages.md]: I'm not sure we need this, I think that is the default?

File: docs/recipes/travis-build-stages.md [hunk 2]
```diff
@@ -1 +1,114 @@
...
+language: node_js
+
+node_js:
+  - 8
+  - 6
+  - 4
+
+os:
+  - linux
+  - osx
```
Comment (gr2m) [inline docs/recipes/travis-build-stages.md]: I'd leave out the `os:` setting for simplicity

File: docs/recipes/travis-build-stages.md [hunk 3]
```diff
@@ -1 +1,114 @@
...
+
+**Advanced configuration**: Running the tests in the `script` step of the `release` stage is not necessary as the previous stage(s) already ran them. To increase speed, the `script` step of the `release` stage can be overwritten to skip the tests. Note that other commands such as build or compilation might still be required.
+
+```yaml
+language: node_js
+
+node_js:
+  - 8
+  - 6
+  - 4
```
Comment (gr2m) [inline docs/recipes/travis-build-stages.md]: Maybe we can leave out `4` already? Either way okay with me

File: docs/recipes/travis-build-stages.md (hunk 1 above)
Comment (pvdlg) [reply docs/recipes/travis-build-stages.md]: Indeed it's the default and we don't need it. I added it for more clarity.
But we can remove it.
Do you think it's more confusing to keep it or to remove it?

File: docs/recipes/travis-build-stages.md (hunk 2 above)
Comment (pvdlg) [reply docs/recipes/travis-build-stages.md]: I removed them in both Travis and Travis build stage

File: docs/recipes/travis-build-stages.md (hunk 3 above)
Comment (pvdlg) [reply docs/recipes/travis-build-stages.md]: I removed Node 4. But we should keep 6 and 8 as the point of the recipe is to show how to test on multiple Node versions

File: docs/recipes/travis-build-stages.md (hunk 1 above)
Comment (pvdlg) [reply docs/recipes/travis-build-stages.md]: I removed it

File: docs/recipes/travis-build-stages.md (hunk 3 above)
Comment (gr2m) [reply docs/recipes/travis-build-stages.md]: good point

**Pull request merged**